ssl/ssl_ciph.c

```diff
@@ -1064,7 +1064,8 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
                  */
                 ERR_raise(ERR_LIB_SSL, SSL_R_INVALID_COMMAND);
                 retval = found = 0;
-                l++;
+                if (ch != '\0')
+                    l++;
```

Comment:
Actually, is this the right fix? This is an error condition, ERR_raise and all, and yet we're breaking out of just the inner loop. It seems we actually just want to break out of the outer loop.

Looking back, it seems we hit this with a fuzzer a while back, but our fix was to make the function immediately fail.
https://boringssl-review.googlesource.com/c/boringssl/+/11421/

Reply:
Returning early seems safer.

Reply:
Yeah. It also seems the function will get confused otherwise. If I'm reading this right, I believe `break` exits to line 1220 ("Ok, we have the rule, now apply it"). That'll could try to execute `CIPHER_SPECIAL` or skip over something and then continue running. We may execute more rules, pile on more errors, before finally returning failure anyway because `retval` is cleared.